src/main/java/com/nerdtranslator/lingueeapibridge/service/impl/NerdTranslatorServiceImpl.java

```diff
@@ -1,22 +1,67 @@
 package com.nerdtranslator.lingueeapibridge.service.impl;
 
+import com.nerdtranslator.lingueeapibridge.data.TranslationData;
+import com.nerdtranslator.lingueeapibridge.service.NaturalLangApiService;
+import com.nerdtranslator.lingueeapibridge.service.TextToSpeechApiService;
 import com.nerdtranslator.lingueeapibridge.service.TranslationApiService;
 import com.nerdtranslator.lingueeapibridge.service.NerdTranslatorService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 @Service
 @RequiredArgsConstructor
 public class NerdTranslatorServiceImpl implements NerdTranslatorService {
     private final TranslationApiService translationApiService;
+    private final TextToSpeechApiService textToSpeechService;
+    private final NaturalLangApiService naturalLanguageApiService;
 
     @Override
-    public String getTranslation(String originalText, String originalLanguage, String targetLanguage) {
-        List<String> translations =
-                translationApiService.fetchDataFromApi("", originalText, originalLanguage, targetLanguage);
-        return null;
+    public TranslationData getTranslation(String originalText, String originalLanguage, String targetLanguage) {
+        TranslationData translationData = new TranslationData();
+        List<String> originWords = getWordsList(originalText);
+        int wordsCount = originWords.size();
+        if (wordsCount == 0) {
+            throw new RuntimeException("wrong input");
+        } else {
+            String singleTranslationFromApi
+                = translationApiService.getSingleTranslationFromApi(originalText, originalLanguage, targetLanguage);
+            translationData.setTranslation(singleTranslationFromApi);
+            List<String> translatedWords = getWordsList(singleTranslationFromApi);
+            int countTranslationWords = translatedWords.size();
+            if (countTranslationWords < 6) {
```

Comment:
It would be better to move all hardcoded values to const variable with clear names